business_objects/embedding.py

```diff
@@ -215,6 +224,39 @@ def get_tensors_and_attributes_for_qdrant(
     return general.execute_all(query)
 
 
+def get_match_record_ids_to_qdrant_ids(
+    project_id: str, embedding_id: str, ids: List[str], limit: int
+) -> List[Any]:
+    # "normal" attributes are stored in qdrant with the record id, embedding lists with the tensor id
+    if not has_sub_key(project_id, embedding_id):
+        return ids
+    query = f"""
+    SELECT et.record_id::TEXT
+    FROM   embedding_tensor et
+    JOIN   UNNEST('{{{",".join(ids)}}}'::uuid[]) WITH ORDINALITY t(id, ord) USING (id)
+    GROUP BY et.record_id
+    ORDER BY MIN(ord)
+    LIMIT {limit}
+    """
+    return [r[0] for r in general.execute_all(query)]
+
+
+def get_qdrant_limit_factor(
+    project_id: str, embedding_id: str, default: int = 1
+) -> int:
+    query = f"""
+    SELECT CEIL(AVG(max_key))::INTEGER
+    FROM (
+        SELECT record_id, MAX(sub_key + 1) max_key
+        FROM embedding_tensor et
+        WHERE project_id = '{project_id}' AND embedding_id = '{embedding_id}'
+        GROUP BY record_id )x """
+    value = general.execute_first(query)
+    if value is None or value[0] is None:
```

Comment:
not sure: can it be that value is `[]`? or is that never the case? if it can be, a check for list length could be reasonable. If it can't be, just resolve :-)

Reply:
value is a scalar through execute_first. the [0] ist access of the first entry first columvalue so -> can't be an empty list